elmslie-android/src/main/java/vivid/money/elmslie/android/renderer/ElmRendererDelegate.kt

```diff
@@ -1,9 +1,114 @@
 package vivid.money.elmslie.android.renderer
 
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.annotation.MainThread
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.withCreated
+import androidx.savedstate.SavedStateRegistryOwner
+import kotlinx.coroutines.launch
+import vivid.money.elmslie.android.elmStore
+import vivid.money.elmslie.core.store.Store
+
 @Suppress("OptionalUnit")
 interface ElmRendererDelegate<Effect : Any, State : Any> {
     fun render(state: State)
     fun handleEffect(effect: Effect): Unit? = Unit
     fun mapList(state: State): List<Any> = emptyList()
     fun renderList(state: State, list: List<Any>): Unit = Unit
 }
+
+/**
+ * The function makes a connection between the store and the lifecycle owner by collecting states and effects
+ * and calling corresponds callbacks.
+ *
+ * Store creates and connects all required entities when given lifecycle reached CREATED state.
+ *
+ * In order to access previously saved state (via [saveState]) in [storeFactory] one must use
+ * SavedStateHandle.get<Bundle>(StateBundleKey)
+ */
+@Suppress("LongParameterList")
+@MainThread
+fun <
+        Event : Any,
+        Effect : Any,
+        State : Any,
+        > ElmRendererDelegate<Effect, State>.connectedElmStore(
+    key: String = this::class.java.canonicalName ?: this::class.java.simpleName,
+    defaultArgs: () -> Bundle = {
+        val args = when (this) {
+            is Fragment -> arguments
+            is ComponentActivity -> intent.extras
+            else -> null
```

Comment:
I'm not sure about this. Yes, it's handy, but at the same time, it's easy to misuse this extension.

Reply:
if you don't use fragments/activity then you don't use this function